Topics/Tech_Stacks/Unity_UI.md

```diff
@@ -0,0 +1,107 @@
+
+Here is a guidance for beginners to start designing their UI in Unity. For more details, please go and check the official [Unity Documentations](https://docs.unity3d.com/2023.3/Documentation/Manual/UIToolkits.html) for the version that you use.
+
+## Start with Canvas
+
+The Canvas is the place where you create and design your UI. It is independent from the actually game scene design and all other Game Objects, and you will find it always on top your other 3D objects or 2D sprite. However, what is cool about Unity UI is taht you can connect the UI with the actually game through game logic scripts.
+
+The Canvas is a Game Object with a Canvas component on it. All the UI element that you created later will be the children of a Canvas.
+
+The Canvas will not be created when you create a new project in Unity, however you can create it by either of the following:
+1. right click anywhere in the hierarchy and select **UI > Canvas**
+2. directly create your first UI element and the Canvas will be automatically created along the way
+
+![Screenshot 2023-11-23 at 10 41 24 PM](https://github.com/arielchen07/learning-software-engineering.github.io/assets/105243552/89523047-c4bc-4cf5-af52-ef8d4fec8b67)
+
+> Tip: If you find the Canvas is not facing towards you, click on the direction arrows on the orientation tool bar to modify the Canvas facing angle.
+<img width="105" alt="Screenshot 2023-11-23 at 7 06 55 PM" src="https://github.com/arielchen07/learning-software-engineering.github.io/assets/105243552/db1220d1-d0ff-4516-a407-50ad58ad9cbc">
+
+## Add UI element
+
+UI elements will be created as the children of the Canvas Game Object.
+
+To create a new UI element, click on the menu **GameObject** or right click anywhere in the hierarchy, hover to **UI** and choose the type of element you want. 
+
+![Screenshot 2023-11-23 at 10 47 23 PM](https://github.com/arielchen07/learning-software-engineering.github.io/assets/105243552/9df58333-1d4c-4462-beda-23d9bed33d89)
+
+Here, we take Text Game Object as an example. By choosing **Text > TextMeshPro**, you should have a new Text Game Object created in the left-side panel of Hierarchy.  
+
+You can create many other types of Game Objects in the simialr fashion.
+
+## Hierarchy: Order of UI element
+
+You probably feel confused when first see Hierarchy in the last section. What is Hierarchy?
+
+In Unity UI, all UI elements will be draw to the Canvas in the same order as they appeared under Canvas Game Object in Hierarvhy window. In other words, if two Game Objects have some overlap, then the latter obejct will be on the top as it will be drawn later.
+
+![Screenshot 2023-11-23 at 12 06 47 AM](https://github.com/arielchen07/learning-software-engineering.github.io/assets/105243552/966981f3-0d0b-48cb-8ac2-56ed2e28ed22)
+
+You can change the order of Game Object by simply drawing them in the Hierarchy window.
```

Comment:
Do you mean dragging the GameObjects in the Hierarchy window to change their order here?

Reply:
Yes you are correct. Do you think i need to rephrase it a bit?

Reply:
yea i think maybe changing the word "drawing" to "dragging" would be more clear here

Reply:
oops I think I mis-typed it. Corrected.